Task3/Regex.java

```diff
@@ -0,0 +1,71 @@
+package Task3;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Regex {
+    public static void main(String[] args) {
+        // Sample input strings
+        String input1 = "app=edi_adapter_converter wingtipsTrace=8faeae6709355291 INFO  OrderCreateClient - action=EDIOrderSent originalFilename=Integration_test_Contract customerName=0005084863 orderUUID=d34149d8-88ab-4791-bb0a-46c96e034200 poNum=Test_TS5155079515 lineCount=3";
+        String input2 = "app=edi_adapter_converter wingtipsTrace=8faeae6709355291 INFO  OrderCreateClient - action=EDIOrderSent originalFilename=Integration_test_Contract customerName=0005084863 orderUUID=d34149d8-88ab-4791-bb0a-46c96e034200 poNum=Test_TS5155079515 lineCount=3";
+        String input3 = "test 2667843 (test_email@griddynamics.com) test 67483 some string";
+        String input4 = "app=edi_adapter_splitter wingtipsTrace=225debfbe6e5fac7 poiFileName=Integration_test_Contract INFO  LogUtils - POI file name: [Integration_test_Contract], total number of orders successfully processed: [2]";
+
+        // 1. Find and return true or false if String contains orderUUID
+        boolean containsOrderUUID1 = containsOrderUUID(input1);
+        System.out.println("Task 1 - Contains orderUUID: " + containsOrderUUID1);
+
+        boolean containsOrderUUID2 = containsOrderUUID(input2);
+        System.out.println("Task 1 - Contains orderUUID: " + containsOrderUUID2);
+
+        // 2. Find and return orderUUID substring from string
+        String orderUUID = findOrderUUID(input2);
+        System.out.println("Task 2 - orderUUID: " + orderUUID);
+
+        // 3. Find and return user email from string
+        String userEmail = findUserEmail(input3);
+        System.out.println("Task 3 - User email: " + userEmail);
+
+        // 4. Find and return how many orders were created from this string
+        int orderCount = findOrderCount(input4);
+        System.out.println("Task 4 - Order count: " + orderCount);
+    }
+
+    // 1. Find and return true or false if String contains orderUUID
+    public static boolean containsOrderUUID(String input) {
```

Comment:
method uses String.matches(), which requires the entire string to match the regex pattern, but the task is to check whether the string contains orderUUUID, not whether the entire string is only orderUUUID.
Method should be configured to search in a string of a given pattern, not to match the whole string to the pattern. 
To do this, you need to use the Pattern and Matcher classes. By compiling the regex into a Pattern and creating a Matcher for the input string, the Matcher.find() method can be used. It scans the string and determines if any part of it matches the regex pattern